README.md

```diff
@@ -521,6 +447,118 @@ If you have a service that produces output, you should define a _non-service_
 script that depends on it, and which exits when the service's output is
 complete.
 
+## Execution cascade
+
+By default, a script always needs to run (or restart in the case of
+[`services`](#services)) if any of its dependencies needed to run, _regardless
+of whether the dependency produced new or relevant output_.
+
+This automatic _cascade_ of script execution is the default behavior because it
+ensures that any _possible_ output produced by a script propagates to all other
+scripts that might depend on it. In other words, Wireit does not assume that the
+`files` array completely describes the inputs to a script.
```

Comment:
These sentences are a bit hard to understand because the context around which script are we talking about. The script that runs as a dependency, or the script who has the dependency.
```suggestion
This automatic _cascade_ of script execution is the default behavior because it
ensures that any _possible_ output produced by a dependent script propagates to all other
scripts that might depend on it. In other words, Wireit does not assume that the
`files` array completely describes the inputs to a script with dependencies.
```